cmd/substreams-sink-noop/main.go

```diff
@@ -166,31 +283,45 @@ func run(cmd *cobra.Command, args []string) error {
 
 	go sinker.Run(ctx)
 
-	zlog.Info("ready, waiting for signal to quit")
-
-	signalHandler, isSignaled, _ := cli.SetupSignalHandler(0*time.Second, zlog)
 	select {
 	case <-signalHandler:
 		go app.Shutdown(nil)
-		break
 	case <-app.Terminating():
-		zlog.Info("run terminating", zap.Bool("from_signal", isSignaled.Load()), zap.Bool("with_error", app.Err() != nil))
-		break
+		zlog.Info("run terminating", zap.Bool("with_error", app.Err() != nil))
 	}
 
 	zlog.Info("waiting for run termination")
 	select {
 	case <-app.Terminated():
+		return app.Err()
 	case <-time.After(30 * time.Second):
 		zlog.Warn("application did not terminate within 30s")
+		return app.Err()
 	}
+}
+
+func computeBlockRangeFromHead(ctx context.Context, blockmetaClient pbbmsrvconnect.BlockClient, reversibleSegmentSize uint64, substreamsSegmentSize uint64, blockRangeArg string) (string, error) {
+	request := connect.NewRequest(&pbbmsrv.Empty{})
 
-	if err := app.Err(); err != nil {
-		return err
+	apiKey := os.Getenv("SUBSTREAMS_API_KEY")
```

Comment:
Once you move this to run() function, you can check this environment variable just once, and maybe return an error that specifies that this environment variable is required especially for Blockmeta. It wouldn't be clear to a user using the substreams_api_token (JWT) why he is required to use the api_key too !